content/docs/19-kubernetes-knowlege-hub/00-how-to/02-how-to-retrieve-images-from-private-registry.md

```diff
@@ -0,0 +1,185 @@
+      labels:
+        app: my-app
+    spec:
+      containers:
+        - name: my-container
+          image: registry.example.com/my-image
+          imagePullSecrets:
+            - name: myregistrykey
+```
+
+## Apply the Deployment Configuration
+
+```bash
+kubectl apply --file deployment.yaml
+```
+
+With this configuration in place, Kubernetes will use the registry credentials in the `myregistrykey` secret to log into the registry and pull the specified image when deploying the application.
+
+## Other Docker Registry Authentication Methods
+
+An alternative way to log into a Docker registry from Kubernetes is by using the command line. 
+
+Authenticate to the private registry. Here’s an example of how to do this:
+
+```bash
+$ kubectl create secret docker-registry <secret-name> \
+    --docker-server=<registry-url> \
+    --docker-username=<username> \
+    --docker-password=<password> \
+    --docker-email=<email>
+```
+
+In the snippet above, **`<secret-name>`** refers to a unique name for the secret, **`<registry-url>`** is the URL of the private registry. Replace the **`<username>`** with the username for authentication and **`<password>`** with the password for authentication. Also, replace **`<email>`**
+ with the email associated with the authentication credentials.
+
+Add the secret created in the previous step to the default service account with the following code.
+
+```bash
+$ kubectl patch serviceaccount default \
+    --port '{"imagePullSecrets": [{"name": "<secret-name>"}]}'
+```
+
+Replace **`<secret-name>`** with the secret created in the previous step.
+
+Once you are authenticated and have added the secret to your default service account, you can use the kubectl command to pull images from the registry and deploy them to your Kubernetes cluster as follows.
+
+```bash
+$ kubectl run <deployment-name> \
+    --image=<registry-url>/<image-name>:<tag> \
+    --port=<container-port>
+```
+
+The line above will create a new deployment using the image specified from the private registry.
+
+# Next Steps
+
+Accessing images from a private registry in Kubernetes can be challenging due to the need to authenticate with the registry. 
+
+To solve this challenge, you have learned how to create a Kubernetes secret with your Docker registry credentials and use it in a Kubernetes deployment configuration. This allows you to pull images from your private registry without restrictions or limitations. 
+
+To learn more about Kubernetes and how to use it to deploy your application, you can visit [our documentation.](https://docs.spectrocloud.com/) You can also read about [how to deploy a stateless frontend application](https://www.notion.so/How-To-Deploy-A-Stateless-Frontend-App-with-Kubernetes-b885ae2307e94ef191a1b713fe29c81f) on Kubernetes or join our [slack channel](https://join.slack.com/t/spectrocloudcommunity/shared_invite/zt-1mw0cgosi-hZJDF_1QU77vF~qNJoPNUQ) to have exciting conversations on Kubernetes with our community members.
```

Comment:
```suggestion
To learn more about Kubernetes and how to use it to deploy your application, check out [Palette's Dev Engine](/devx/apps/deploy-app) and how it can reduce the challenges often encountered with deploying apps to Kubernetes. You can also read about [how to deploy a stateless frontend application](/kubernetes-knowlege-hub/how-to/deploy-stateless-frontend-app) on Kubernetes or join our [slack channel](https://join.slack.com/t/spectrocloudcommunity/shared_invite/zt-1mw0cgosi-hZJDF_1QU77vF~qNJoPNUQ). Learn from other Kubernetes users and get to know fellow community members.
```

FIxed the URLs that were pointing back to Notion. Minor language cleanup at the end. And lastly, instead of pointing generically to our docs. We can point the target audience to Palette PDE.